### DISCUSSION TITLE
Batch spell check? And OCR settings

### DISCUSSION THREAD

**Discussion by DrReddP** [SubtitleEdit/subtitleedit#4972] [Q&A]
1. Is there a way to spell check multiple files at once? I'm not seeing an option in batch.

2. In the attached image, in the red box, there used to be font color options for OCR. They're gone now, and because I can't turn off custom colors, I'm getting way more OCR errors. Has this setting moved somewhere?
![1](https://user-images.githubusercontent.com/82276985/114274454-6faef280-99ec-11eb-8c62-ff7197c33bf4.png)

**Comment by niksedk** [comment]
1) Sorry, no. Why would that be useful?

2) The custom palette is only for vosub (sub/idx) files. Sometimes the font color is wrong in vobsub, e.g. due to missing/bad idx file.

**Comment by DrReddP** [comment]
The reason batch spell check would be helpful.

Right now I'm working the the subtitles for a sci-fi show called Stargate
SG-1. There are a lot of "alien" words used that I don't want to add to my
dictionary, so I just hit "skip all" when spell check comes to them.

 Right now, I have to repeat that action for the same word 214 times
because there are 214 episodes. If I was able to spell check them all at
once, I could hit "skip all" one time and it would never ask me about that
word again. Also the "use always" and "change all" buttons would only have
to be used once instead of over and over for each episode.

Next question: how does "multiple word replace" work in batch mode? When I
selected that I assumed it would ask me what words to replace and with what
but it didn't. How do I know and control which words it's replacing?

**Reply by OmrSi** [reply]
Why not add the alien words to your user dictionary and then just clear your user dictionary?

You can't control what multiple replace replaces.
If you enable the option, it will apply the replacements without asking.

**Comment by DrReddP** [comment]
But how do I know what's being replaced with what?

And the reason I can't add the alien words to my dictionary and then clear
my dictionary is because there are other words that I need to add that I
want to stay there. Words like "stargate". Additionally, that doesn't fix
the fact that I have to open and spell check 200 different files when I
could just do it once, nor does it resolve having to hit "skip all" or
"change all" for every single one.

To be clear I'm not complaining, I was just asking if it was a thing. And
maybe suggesting it as a feature.

**Comment by niksedk** [comment]
General names should be added to your names list (Spell check - Add word to names list).
Can be fixed in batch via "Redo casing".

A few years back I used the "Edit - Multiple replace" to correct a TV show where some names/words required another casing. 
Make a new group with your fixes - and you can fix 200 episodes in less than a minute via batch convert.

PS: Saw SG-1 when it came out... was a big fan.

**Comment by DrReddP** [comment]
I'm not talking about names. I'm talking about common words.

As to the "make a new group with your fixes" I have no idea what that means.

And again, how do you know what's being replaced when you do "multiple
replace"? What's getting replaced with what?

**Comment by DrReddP** [comment]
I finally figured out the multiple replace. I didn't know there was a
"settings" button for it

**Comment by DrReddP** [comment]
Ok next question: when setting up "multiple replace", there are casing
options (see attached), but no "whole word only" option like there is in
the regular Subtitle Edit.

For example, I used batch to capitalize the word "general", but it also
capitalized all the occurrences of "generalize". How do I do "whole word
only"?

**Reply by OmrSi** [reply]
Use the `Regular expression` option.
Replace `\bgeneral\b` with `General`.
It's case-sensitive by default.
If you want it to be case-insensitive, you can replace  `(?i)\bgeneral\b` with `General`.

**Reply by OmrSi** [reply]
Also, I believe attachments from the email don't show up in the site.

**Reply by DrReddP** [reply]
Here, sorry. There's already a case sensitive option
![20210412_145057](https://user-images.githubusercontent.com/82276985/114448627-ca3d7f80-9ba1-11eb-80a5-a27db19d05a9.png)

**Reply by DrReddP** [reply]
Oh I see, I can't select both so I need the stuff you told me. Ty

**Comment by DrReddP** [comment]
Ok next question: when setting up "multiple replace", there are casing
options (see attached), but no "whole word only" option like there is in
the regular Subtitle Edit.

For example, I used batch to capitalize the word "general", but it also
capitalized all the occurrences of "generalize". How do I do "whole word
only"?

**Reply by niksedk** [reply]
Please, only *one* subject for each discussion, thx :)
